src/ports/web/api/events/export.rs

```diff
@@ -0,0 +1,176 @@
+use super::*;
+
+#[get("/export/events.csv?<query..>")]
+pub fn csv_export(
+    connections: sqlite::Connections,
+    search_engine: tantivy::SearchEngine,
+    auth: Auth,
+    query: usecases::EventQuery,
+) -> result::Result<Content<String>, AppError> {
+    let events = query_events(connections, search_engine, auth, query)?;
+
+    let records: Vec<_> = events.map(adapters::csv::EventRecord::from).collect();
+
+    let buff: Vec<u8> = vec![];
+    let mut wtr = csv::Writer::from_writer(buff);
```

Comment:
I have never seen the abbreviation `wtr` before. Please use complete words unless there it is commonly known and used abbreviation, e.g. `err` or `res` in Rust.